spec/controllers/saml_idp_controller_spec.rb

```diff
@@ -1673,217 +1627,155 @@ def name_id_version(format_urn)
-        IdentityLinker.new(user, service_provider).link_identity
-        user.identities.last.update!(verified_attributes: ['email'])
-        generate_saml_response(user, auth_settings)
+      shared_examples_for 'sends the UUID' do
+        it 'sends the UUID' do
+          generate_saml_response(user, auth_settings)
 
-        expect(response.status).to eq(200)
+          expect(response.status).to eq(200)
 
-        analytics_hash = {
-          success: true,
-          errors: {},
-          nameid_format: Saml::Idp::Constants::NAME_ID_FORMAT_PERSISTENT,
-          authn_context: request_authn_contexts,
-          authn_context_comparison: 'exact',
-          requested_ial: Saml::Idp::Constants::IAL1_AUTHN_CONTEXT_CLASSREF,
-          service_provider: 'http://localhost:3000',
-          endpoint: "/api/saml/auth#{path_year}",
-          idv: false,
-          finish_profile: false,
-          request_signed: true,
-          matching_cert_serial: saml_test_sp_cert_serial,
-        }
+          expect(name_id.attributes['Format'].value).
+            to eq(Saml::Idp::Constants::NAME_ID_FORMAT_PERSISTENT)
 
-        expect(@analytics).to have_received(:track_event).
-          with('SAML Auth', analytics_hash)
+          expect(name_id.children.first.to_s).to eq(user.last_identity.uuid)
+        end
       end
 
-      it 'defaults to email when added to issuers_with_email_nameid_format' do
-        auth_settings = saml_settings(
-          overrides: {
-            issuer: sp1_issuer,
-            name_identifier_format: nil,
-          },
-        )
-        ServiceProvider.
-          find_by(issuer: auth_settings.issuer).
-          update!(email_nameid_format_allowed: true)
-        IdentityLinker.new(user, sp1).link_identity
-        user.identities.last.update!(verified_attributes: ['email'])
-        generate_saml_response(user, auth_settings)
+      shared_examples_for 'sends the email' do
+        it 'sends the email' do
+          generate_saml_response(user, auth_settings)
 
-        expect(response.status).to eq(200)
+          expect(response.status).to eq(200)
 
-        analytics_hash = {
-          success: true,
-          errors: {},
-          nameid_format: Saml::Idp::Constants::NAME_ID_FORMAT_EMAIL,
-          authn_context: request_authn_contexts,
-          authn_context_comparison: 'exact',
-          requested_ial: Saml::Idp::Constants::IAL1_AUTHN_CONTEXT_CLASSREF,
-          service_provider: auth_settings.issuer,
-          endpoint: "/api/saml/auth#{path_year}",
-          idv: false,
-          finish_profile: false,
-          request_signed: true,
-          matching_cert_serial: saml_test_sp_cert_serial,
-        }
+          expect(name_id.attributes['Format'].value).
+            to eq(Saml::Idp::Constants::NAME_ID_FORMAT_EMAIL)
 
-        expect(@analytics).to have_received(:track_event).
-          with('SAML Auth', analytics_hash)
```

Comment:
I agree with the previous comment that we should keep some of the assertions around analytics logging in these specs.

I agree that asserting an entire payload migth be too much, but we have a pattern of logs of  `hash_including` that let us assert against a subset

```ruby
expect(@analytics).to have_logged_event('SAML Auth', hash_including(events_subset))
```

I think that's particularly relevant this week, because the deploy we rolled back was due to us logging the wrong information, so if we intend to fix the logging, it would be good to have specs around said logging